app/models/concerns/sentiment_analysis_helper.rb

```diff
@@ -0,0 +1,40 @@
+module SentimentAnalysisHelper
+  extend ActiveSupport::Concern
+
+  def opening_sentiments
+    records = incoming_messages.first(average_message_count)
+    average_sentiment(records)
+  end
+
+  def closing_sentiments
+    records = incoming_messages.last(average_message_count)
+    average_sentiment(records)
+  end
+
+  def average_sentiment(records)
+    {
+      label: average_sentiment_label(records),
+      score: average_sentiment_score(records)
+    }
+  end
+
+  private
+
+  def average_sentiment_label(records)
+    value = records.pluck(:sentiment).sum { |a| a['value'].to_i }
+    value.negative? ? 'negative' : 'positive'
+  end
+
+  def average_sentiment_score(records)
```

Comment:
This looks good for now, lets see how this works out with real conversations. If it turns out this needs tweaking, maybe we can have a weighted average, say we have 5 conversations, for the opening sentiment the first message will have more weight, and subsequent messages will have lesser and lesser weight

We can use an exponential decay algorithm

```rb
decay_factor = 0.9  # This value should be between 0 and 1
weights = Array.new(n) { |i| (decay_factor ** (n - i - 1)) }
```

For now we can let this implementation stay

Reply:
> We can use an exponential decay algorithm

Yes, this sounds good. This is for the cloud customer now to check how it turns out. Then we can think about this.